kernel/src/process_system.c

```diff
@@ -27,6 +37,177 @@ __attribute__((always_inline)) uint64_t generate_pid() {
     return pid++;
 }
 
+__attribute__((naked)) void context_switch_handler() {
+    asm volatile(
+        // Save rax on stack
+        "push %rax\n"
+        "push %rbx\n"
+
+        // Put g_process_queue ptr in rax
```

Comment:
you could send eoi here to avoid duplicate code. Since rax is immediately overwritten by the lea afterwards, you dont need to push/pop any registers either.

Reply:
But shouldn't the EOI be sent at the end of the interrupt?

Reply:
We need to push rax and rbx to not overwrite the process register values while using them.